File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/HashHmacFile.java
package com.onevgo.functions;

import cn.hutool.crypto.digest.HMac;

import java.io.File;
import java.io.IOException;

public class HashHmacFile {
    public static String hashHmacFile(String algo, String filename, byte[] key) {
        return new HMac(algo, key).digestHex(new File(filename));
    }

    public static void main(String[] args) throws IOException {
        FilePutContents.filePutContents("test.txt", "The quick brown fox jumped over the lazy dog.");
        System.out.println(hashHmacFile("HMACMD5", "test.txt", "secret".getBytes()));
    }
}

File: 01_Language/01_Functions/python/hash_hmac_file.py
# coding: utf-8

import hmac


def hash_hmac_file(alog, filename, key):
    with open(filename, mode="rb") as f:
        h = hmac.new(key, None, alog)
        while True:
            data = f.read(8192)
            if not data:
                break
            h.update(data)
        return h.hexdigest()


if __name__ == '__main__':
    import file_put_contents

    file_put_contents.file_put_contents("test.txt", "The quick brown fox jumped over the lazy dog.")
    print(hash_hmac_file("md5", "test.txt", b"secret"))
